drizzlepac/haputils/svm_quality_analysis.py

```diff
@@ -516,6 +516,304 @@ def compare_ra_dec_crossmatches(hap_obj, json_timestamp=None, json_time_since_ep
 # ------------------------------------------------------------------------------------------------------------
 
 
+def compare_interfilter_crossmatches(total_obj_list, json_timestamp=None, json_time_since_epoch=None,
+                                     log_level=logutil.logging.NOTSET):
+    """Compare X, Y positions of cross matched sources from different filter-level products
+
+    Parameters
+    ----------
+    total_obj_list : list
+        list of drizzlepac.haputils.Product.TotalProduct objects to process
+
+    json_timestamp: str, optional
+        Universal .json file generation date and time (local timezone) that will be used in the instantiation
+        of the HapDiagnostic object. Format: MM/DD/YYYYTHH:MM:SS (Example: 05/04/2020T13:46:35). If not
+        specified, default value is logical 'None'
+
+    json_time_since_epoch : float
+        Universal .json file generation time that will be used in the instantiation of the HapDiagnostic
+        object. Format: Time (in seconds) elapsed since January 1, 1970, 00:00:00 (UTC). If not specified,
+        default value is logical 'None'
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the
+        .log file. Default value is 'NOTSET'.
+
+    Returns
+    -------
+    Nothing
+    """
+    # Initiate logging!
+    log.setLevel(log_level)
+
+    # Check to make sure there's at last 2 filter-level products. If not, return.
+    num_filter_prods = 0
+    for total_obj in total_obj_list:
+        num_filter_prods += len(total_obj.fdp_list)
+    if num_filter_prods == 1:
+        log.error("At least 2 filter-level products are necessary to perform analysis of inter-filter "
+                  "cross matched sources. Only 1 filter-level product was found. ")
+        return
+    elif num_filter_prods == 0:
+        log.error("At least 2 filter-level products are necessary to perform analysis of inter-filter "
+                  "cross matched sources. No filter-level products were found. ")
+        return
+    else:
+        log.info("Found {} filter-level products for use in analysis of inter-filter cross matched source"
+                 " positions:".format(num_filter_prods))
+        ctr = 1
+        for total_obj in total_obj_list:
+            for filt_obj in total_obj.fdp_list:
+                log.info("{}: {}".format(ctr, filt_obj.drizzle_filename))
+                ctr += 1
+
+        filtobj_dict = {}
+        xmatch_ref_imgname = None
+        max_sources = 0
+
+        for total_obj in total_obj_list:
+            for filt_obj in total_obj.fdp_list:
+                log.info("{} {} {}".format(">"*20, filt_obj.drizzle_filename, "<"*20))
```

Comment:
You could include the `ctr` from the previous loop in here and add it to this log line with something like:
    "{} Filter #{}: {}  {}".format(">"*20, ctr, filt_obj.drizzle_filename, '<'*20)

Functionally, this would not change anything, but it would limit the number of lines in the log file and still provide the same information while removing an extra loop over the same information.